components/bin/copy

```diff
@@ -0,0 +1,89 @@
+#! /usr/bin/env node
+
+/*************************************************************
+ *
+ *  Copyright (c) 2018 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Copies specified files to the distribution directory
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * The amount of space for each level of indentation
+ */
+const INDENT = '  ';
+
+/**
+ * The configuration data for the copy operation
+ */
+const config = JSON.parse(fs.readFileSync(process.argv[2] || 'copy.json'));
+
+/**
+ * Get the directory for node modules (either the parent of the MathJax3 directory,
+ * or the MathJax3 node_modules directory, if it exists).
+ */
+const parent = path.resolve(__dirname, '..', '..');
+const nodeDir = (dir => (fs.existsSync(dir) ? dir : path.resolve(parent, '..')))(path.join(parent, 'node_modules'));
+
+/**
+ * Copy a file or directory tree
+ *
+ * @param {string} from     The directory to copy from
+ * @param {string} to       The directory to copy to
+ * @param {string} name     The name of the file or directory to copy
+ * @param {string} space    The indentation for output
+ */
+function copyFile(from, to, name, space) {
+    !fs.existsSync(to) && makeDir(to);
+    const copy = path.resolve(from, name);
+    const dest = path.resolve(to, name);
+    if (fs.lstatSync(copy).isDirectory()) {
+        console.log(space + name + '/');
+        for (const file of fs.readdirSync(copy)) {
+            copyFile(copy, dest, file, space + INDENT);
+        }
+    } else {
+        console.log(space + name);
+        fs.copyFileSync(copy, dest);
+    }
+}
+
+/**
+ * Recursively make any needed directories
+ *
+ * @param {string} dir   The directory relative to the library directory
+ */
+function makeDir(dir) {
+    if (fs.existsSync(dir)) return;
+    makeDir(path.dirname(dir));
+    fs.mkdirSync(dir);
```

Comment:
Should we not just use the node facility for `mkdir -p`:
```suggestion
    fs.mkdirSync(dir, {recursive: true});
```

Reply:
I was not aware of this option (and I thought I had looked, but must have missed it).  That makes this function unnecessary, so I've modified the code to remove the function and use 1fs.mkdirSync(to, {recursive: true});` in place of `makeDir(to)` earlier.